code/game/objects/items/weapons/melee/misc.dm

```diff
@@ -44,4 +44,57 @@
 		var/mob/living/carbon/human/H = src.loc
 		H.update_inv_l_hand(0)
 		H.update_inv_r_hand()
-	..()
\ No newline at end of file
+	..()
+
+/obj/item/weapon/melee/cursedblade
+	name = "crystal blade"
+	desc = "The red crystal blade's polished surface glints in the light, giving off a faint glow."
+	icon_state = "soulblade"
+	slot_flags = SLOT_BELT | SLOT_BACK
+	force = 30
+	throwforce = 10
+	w_class = ITEMSIZE_NORMAL
+	sharp = 1
+	edge = 1
+	attack_verb = list("attacked", "slashed", "stabbed", "sliced", "torn", "ripped", "diced", "cut")
+	hitsound = 'sound/weapons/bladeslice.ogg'
+	can_speak = 1
+	var/mob/living/carbon/brain/brainmob = null //The curse of the sword is that it has someone trapped inside.
+
+
+/obj/item/weapon/melee/cursedblade/handle_shield(mob/user, var/damage, atom/damage_source = null, mob/attacker = null, var/def_zone = null, var/attack_text = "the attack")
+	if(default_parry_check(user, attacker, damage_source) && prob(50))
+		user.visible_message("<span class='danger'>\The [user] parries [attack_text] with \the [src]!</span>")
+		playsound(user.loc, 'sound/weapons/punchmiss.ogg', 50, 1)
+		return 1
+	return 0
+
+/obj/item/weapon/melee/cursedblade/New()
+	src.brainmob = new(src)
+	src.brainmob.add_language(LANGUAGE_GALCOM)
+	src.brainmob.loc = src
+	src.brainmob.container = src
+	src.brainmob.stat = 0
+	src.brainmob.silent = 0
+	src.brainmob.name = "cursed sword"

```

Comment:
Where's the framework part?

Reply:
In say.dm. For a brain to talk now requires the container its in to have a can_speak variable. Previously it checked if the brain was in an MMI and called it a day.

Reply:
The only things that have can_say set by default are MMIs (which includes posibrains and drone chips) and this sword, but admin magic can be used to make other things capable of speech.

Reply:
> Adds a framework to allow ghosts to 'haunt' an object and let them speak through it.

I mean, I interpret this as adding a mechanism by which ghosts can join into objects. Is that not the case? Has to be adminbused in?

Reply:
That's already the ghost_pods. The particular ghost pod for this sword I am currently working on. But yes, the framework I was talking about is mostly for adminbuse. Perhaps that was a bit misleading. I'll reword that.